k8s/plugin/src/resources/upgrade.rs

```diff
@@ -0,0 +1,328 @@
+use crate::{
+    constant::{
+        UPGRADE_CONTROLLER_DEPLOYMENT, UPGRADE_IMAGE, UPGRADE_OPERATOR_CLUSTER_ROLE,
+        UPGRADE_OPERATOR_CLUSTER_ROLE_BINDING, UPGRADE_OPERATOR_SERVICE,
+        UPGRADE_OPERATOR_SERVICE_ACCOUNT,
+    },
+    resources::objects,
+};
+
+use anyhow::Error;
+use k8s_openapi::api::{
+    apps::v1::Deployment,
+    core::v1::{Service, ServiceAccount},
+    rbac::v1::{ClusterRole, ClusterRoleBinding},
+};
+use kube::{
+    api::{Api, DeleteParams, ListParams, PostParams},
+    core::ObjectList,
+    Client,
+};
+use std::time::Duration;
+
+/// The types of resources that support the upgrade operator.
+#[derive(clap::Subcommand, Debug)]
+pub enum UpgradeOperator {
+    /// Install, Uninstall upgrade resources.
+    UpgradeOperator,
+}
+
+/// K8s resources needed for upgrade operator
```

Comment:
missing termination `.` here and in other doc comments below.

Reply:
This needs to be resolved.